File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    answer = []
    
    i = 0
    while (i < len(arr) and len(answer) < k):
        if arr[i] not in answer:
            answer.append(arr[i])
        i += 1
    
    while (k > len(answer)):
        answer.append(-1)
    
    return answer

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, int k) {
    vector<int> answer;
    bool check[100001] = {false};
    
    for (int i = 0; i < arr.size() && k > 0; i++) {
        if (check[arr[i]])
            continue;
        answer.push_back(arr[i]);
        check[arr[i]] = true;
        k--;
    }
    
    while (k--)
        answer.push_back(-1);
    
    return answer;
}